pyproject.toml

```diff
@@ -0,0 +1,50 @@
+[tool.poetry]
+name = "onionshare"
+version = "2.2"
+description = "OnionShare lets you securely and anonymously send and receive files. It works by starting a web server, making it accessible as a Tor onion service, and generating an unguessable web address so others can download files from you, or upload files to you. It does _not_ require setting up a separate server or using a third party file-sharing service."
+authors = ["Micah Lee <micah.lee@theintercept.com>"]
+license = "GPLv3+"
+
+[tool.poetry.dependencies]
+python = "^3.7"
+altgraph = "0.16.1"
+certifi = "2019.9.11"
+chardet = "3.0.4"
+Click = "7.0"
+Flask = "1.1.1"
+Flask-HTTPAuth = "3.3.0"
+future = "0.17.1"
+idna = "2.8"
+itsdangerous = "1.1.0"
+Jinja2 = "2.10.1"
+macholib = "1.11"
+MarkupSafe = "1.1.1"
+pefile = "2019.4.18"
+pycryptodome = "3.9.0"
+PyQt5 = "5.13.1"
+PyQt5-sip = "4.19.19"
+PySocks = "1.7.0"
+requests = "2.22.0"
+stem = "1.7.1"
+urllib3 = "1.25.3"
+Werkzeug = "0.15.6"
+watchdog = "0.9.0"
+psutil = "5.6.7"
+
+[tool.poetry.dev-dependencies]
+atomicwrites = "1.3.0"
+attrs = "19.1.0"
+more-itertools = "7.2.0"
+pluggy = "0.13.0"
+py = "1.8.0"
+pytest = "5.1.2"
+pytest-faulthandler = "2.0.1"
+pytest-qt = "3.2.2"
+six = "1.12.0"
+urllib3 = "1.25.3"
```

Comment:
None of these are tied to any specific version, they can all have `^` -- and while you're at it, I would be fine bumping all of the versions to their newest versions, so long as the tests pass

Reply:
You can update to all the latest versions with `poetry update`

Reply:
Actually, while troubleshooting in #1079 I discovered that if you set the version for all of the dependencies to `*`, and then run `poetry install` and `poetry update`, it updates all dependencies in `poetry.lock`, and fixes the python 3.8/stem problem.

Reply:
Changed everything to `*` and then did a `poetry update` to update the `poetry.lock`